### PR TITLE
Adding DayPickerSingleDateController to react-dates

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/airbnb/react-dates/blob/536110b5658c060b0ff1a60a5a088ef29b629558/src/components/DayPickerSingleDateController.jsx
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-dates/index.d.ts [hunk 1]
```diff
@@ -275,7 +275,63 @@ declare namespace ReactDates {
...
+        isDayHighlighted?: (day: any) => boolean,
+        
+        // DayPicker props
+        renderMonth?: Function,
+        enableOutsideDays?: boolean,
+        numberOfMonths?: number,
+        orientation?: ScrollableOrientationShape,
+        withPortal?: boolean,
+        initialVisibleMonth?: Function,
+        firstDayOfWeek? : 0 | 1 | 2 | 3 | 4 | 5 | 6,
```
Comment (ArturAmpilogov) [inline types/react-dates/index.d.ts]: Please move this to DayOfWeekShape
https://github.com/airbnb/react-dates/blob/master/src/components/DayPickerSingleDateController.jsx#L54
the rest is good, thank you

type DayOfWeekShape = 0 | 1 | 2 | 3 | 4 | 5 | 6;
https://github.com/airbnb/react-dates/blob/master/src/constants.js#L28

**Review by ArturAmpilogov — CHANGES_REQUESTED** [review]

**Review by Artur-A — APPROVED** [review]

**Comment by caltonji** [comment]
@Artur-A Would you mind taking a second look at this?

**Comment by caltonji** [comment]
@ArturAmpilogov @typescript-bot  Do you know why this still has revision needed?

**Comment by ArturAmpilogov** [comment]
Hi @caltonji, an additional revision by a Microsoft employee is required. Usually, it is done during 2-5 days depending on the team workload.

**Comment by ArturAmpilogov** [comment]
@caltonji that is a good point but not sufficient here :)

@sheetalkamat can you please check or ping other admins?

**Comment by minestarks** [comment]
@caltonji Could you address the merge conflict? I'll merge once you're done. Thanks!

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has merge conflicts and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**